rich/markdown.py

```diff
@@ -302,23 +299,33 @@ def render_number(
             yield new_line
 
 
+class Link(TextElement):
+    @classmethod
+    def create(cls, markdown: "Markdown", token: Any) -> "MarkdownElement":
+        return cls(token.content, token.attrs.get("href"))
+
+    def __init__(self, text: str, href: str):
+        self.text = Text(text)
+        self.href = href
+
+
 class ImageItem(TextElement):
     """Renders a placeholder for an image."""
 
     new_line = False
 
     @classmethod
-    def create(cls, markdown: "Markdown", node: Any) -> "MarkdownElement":
+    def create(cls, markdown: "Markdown", token: Any) -> "MarkdownElement":
         """Factory to create markdown element,
 
         Args:
             markdown (Markdown): The parent Markdown object.
-            node (Any): A node from Pygments.
+            token (Any): A token from markdown-it.
 
         Returns:
             MarkdownElement: A new markdown element
         """
-        return cls(node.destination, markdown.hyperlinks)
+        return cls(token.attrs.get("src"), markdown.hyperlinks)
```

Comment:
I'm a bit confused here. Looks like this may potentially call `cls` with `None`, which doesn't look allowed from the `__init__` below. Surprised typing didn't pick that up.

Reply:
It's because the signatures were typed as `Any`, and I never changed them.  I've updated them to ensure we get more help from `mypy` (and fixed all the corresponding issues, including the one you mentioned).